python/ray/__init__.py

```diff
@@ -4,6 +4,12 @@
 
 import os
 import sys
+
+if "pyarrow" in sys.modules:
+    raise Exception("Ray must be imported before pyarrow because Ray requires "
```

Comment:
Should this be a `RuntimeError`? `ImportError`?

Reply:
Great point, should be fixed now.